File: ios/GlassesControl/GlassesControl/Glasses/GlassesControlScreen.swift
//Copyright (c) 2020 Everysight LTD

import Foundation
import NativeEvsKit

class GlassesControlScreen : Screen {
    private let text = Text()
    override func onCreate() {
        text
            .setText(text: "Hello World")
            .setResource(font: Font(stockFont: .medium))
            .setTextAlign(align: Align.center)
        text
            .setX(x: getWidth()/2)
            .setY(y: getHeight()/2)
            .setForegroundColor(color: EvsColor.green.rgba)
        
        add(uiElement: text)
    }
    
    override func onTouch(touchDirection: TouchDirection) -> Bool {
        text.setText(text: "Touch \(touchDirection)")
        return false
    }
}

File: android/GlassesControl/src/main/java/com/everysight/sdk/samples/glassescontrol/GlassesControlScreen.kt
package com.everysight.sdk.samples.glassescontrol

import UIKit.app.Screen
import UIKit.app.data.Align
import UIKit.app.data.EvsColor
import UIKit.app.data.TouchDirection
import UIKit.app.resources.Font
import UIKit.widgets.Text


class GlassesControlScreen:Screen() {
    internal val text = Text()
    override fun onCreate() {

        text.setText("Hello World").setResource(Font.StockFont.Medium).setTextAlign(Align.center)
        text.setX(getWidth()/2).setY(getHeight()/2).setForegroundColor(EvsColor.Green.rgba)
        add(text)
    }

    override fun onTouch(touch: TouchDirection): Boolean {
        super.onTouch(touch)
        text.setText("Touch: $touch")
        return false
    }
}